### PR TITLE
feat: schema-driven saved view summary rendering

### PR DESCRIPTION
**Pull request by ghost**
> Opened by @roomote-v0 on behalf of Nathan Horstmeier

Replace naive key-capitalization and `String(value)` rendering in ProfileBar saved view summaries with schema-driven formatting via a new `buildFilterSummary()` utility.

## Changes

- **`src/filters/filterState.js`** -- Add `buildFilterSummary(filters, schema)` that walks the filter schema to produce structured `{ key, label, type, displayValues }` items with type-aware formatting (option label lookup for select/multi-select, quoted text, formatted date ranges, Yes/No for booleans, pillLabel overrides)
- **`src/WorksCalendar.tsx`** -- Pass `schema` prop to `<ProfileBar>` and include `schema` + `buildFilterSummary` helper in the `renderSavedViewsBar` callback payload
- **`src/ui/ProfileBar.jsx`** -- Accept `schema` prop, replace hardcoded `buildSummary()` and `FilterSummary` with calls to `buildFilterSummary()`, remove the old helper
- **`src/filters/__tests__/filterState.test.js`** -- Add 26 tests covering all field types, edge cases (empty filters, unknown keys, schema mismatch), pillLabel overrides, option label resolution, and schema factory functions

## Fixes

- `dateRange` objects no longer render as `[object Object]`
- `boolean` filters show "Yes"/"No" instead of "true"/"false"
- `select` fields resolve option labels instead of showing raw values
- Field labels come from `schema.label` (e.g. "Date") instead of naive capitalize ("Daterange")
- `pillLabel` overrides are now respected in summaries

All 642 existing tests pass with zero regressions.

---

[View task on Roo Code Cloud](https://app.roocode.com/cloud-jobs/lzhMwLoq?utm_source=github-comment&utm_medium=link&utm_campaign=code.task)

### CODE REVIEW

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task](https://app.roocode.com/cloud-jobs/OZb27rV0?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Clean PR overall. The new `buildFilterSummary()` utility is well-structured, handles all field types properly, and has solid test coverage. One bug found:

- [ ] `formatDate()` parses date-only ISO strings as UTC but renders in local time, causing off-by-one date display for users west of UTC ([comment](https://github.com/natehorst240-sketch/CalendarThatWorks/pull/95#pullrequestreview-4111767467))

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=natehorst240-sketch/CalendarThatWorks&prNumber=95&commentId=4250223685).</sub>

File: src/filters/filterState.js [hunk 1]
```diff
@@ -98,3 +98,158 @@ export function buildActiveFilterPills(filters, schema = DEFAULT_FILTER_SCHEMA)
...
+
+/**
+ * Format a date value (Date object or ISO string) as a short readable string.
+ */
+function formatDate(d) {
+  if (!d) return null;
+  const date = d instanceof Date ? d : new Date(d);
+  if (isNaN(date.getTime())) return String(d);
+  return date.toLocaleDateString(undefined, { month: 'short', day: 'numeric', year: 'numeric' });
+}
```
Comment (ghost) [inline src/filters/filterState.js:209]: `new Date('2026-04-01')` parses date-only ISO strings as UTC midnight. `toLocaleDateString(undefined, ...)` then renders in the user's local timezone. For anyone west of UTC (most of the Americas), `2026-04-01` would display as "Mar 31, 2026". Passing `{ timeZone: 'UTC' }` in the options object fixes it without changing the output for UTC+ users.

```suggestion
function formatDate(d) {
  if (!d) return null;
  const date = d instanceof Date ? d : new Date(d);
  if (isNaN(date.getTime())) return String(d);
  return date.toLocaleDateString(undefined, { month: 'short', day: 'numeric', year: 'numeric', timeZone: 'UTC' });
}
```

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=natehorst240-sketch/CalendarThatWorks&prNumber=95&fixId=01c6420efc) or mention @roomote and request a fix.</sub>

**Comment by ghost** [comment]
> **Rooviewer**  Clock &nbsp;&nbsp;[See task](https://app.roocode.com/cloud-jobs/ga5geI87?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review)

Well-structured PR with solid test coverage. One bug found:

- [ ] `formatDate()` parses date-only ISO strings as UTC but renders in local time, causing off-by-one date display for users west of UTC ([comment](https://github.com/natehorst240-sketch/CalendarThatWorks/pull/95#pullrequestreview-4111826182))

<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=natehorst240-sketch/CalendarThatWorks&prNumber=95&commentId=4250290371).</sub>

File: src/filters/filterState.js (hunk 1 above)
Comment (ghost) [inline src/filters/filterState.js:209]: `formatDate()` parses date-only ISO strings (`'2026-04-01'`) as UTC midnight, but `toLocaleDateString(undefined, ...)` formats in the browser's local timezone. For users west of UTC (most of the Americas), this causes an off-by-one: April 1 renders as "Mar 31, 2026". Adding `timeZone: 'UTC'` to the options fixes it without affecting UTC+ locales.

```suggestion
function formatDate(d) {
  if (!d) return null;
  const date = d instanceof Date ? d : new Date(d);
  if (isNaN(date.getTime())) return String(d);
  return date.toLocaleDateString(undefined, { month: 'short', day: 'numeric', year: 'numeric', timeZone: 'UTC' });
}
```

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=natehorst240-sketch/CalendarThatWorks&prNumber=95&fixId=d0a70655c1) or mention @roomote and request a fix.</sub>
